src/main/java/uk/ac/cam/cl/dtg/isaac/api/QuizFacade.java

```diff
@@ -836,7 +836,7 @@ public final Response abandonQuizAttempt(@Context final HttpServletRequest httpS
             QuizAttemptDTO quizAttempt = this.quizAttemptManager.getById(quizAttemptId);
```

Comment:
Not quite releated to this PR, but what happens here is there is no attempt with that ID? Ideally we'd give the same error message as for the user ID mismatch to hide the internal details, does this throw the exception that's caught with an unhelpful error message below? Because otherwise there's an NPE if `quizAttempt` is null?

This probably applies more widely than this one, it's just it took me reading this far to think of it.

Reply:
This one throws a `SegueDatabaseException` so it's caught by the appropriate clause.